pastis/e2e_simulators/generic_segmented_telescopes.py

```diff
@@ -787,14 +788,15 @@ def _transform_harris_mode(values, xrot, yrot, points, seg_evaluated, seg_num):
                 mode_set_per_segment.extend([ZB, ZC, ZD])
 
             harris_base.append(mode_set_per_segment)
-
         # Create full mode basis of selected Harris modes on all segments
         harris_base = np.asarray(harris_base)
         self.n_harris_modes = harris_base.shape[1]
         harris_base = harris_base.reshape(self.nseg * self.n_harris_modes, pup_dims[0] ** 2)
         harris_mode_basis = hcipy.ModeBasis(np.transpose(harris_base), grid=self.pupil_grid)
 
         self.harris_sm = hcipy.optics.DeformableMirror(harris_mode_basis)
+        if return_mode:
+            return mode_set_per_segment
```

Comment:
Why did you end up returning this? I would rather not touch any of the existing scripts in this PR to avoid breaking things and just add the notebooks. We can do anything else in follow-up work.

Reply:
I believe to plot the harris segment level aberrations, we need to use a major chunk of the code written inside the function create_segmented_harris_mirror especially from L26 to L88. I was using these lines in my notebook. 
You said notebooks were getting too long, difficult to read and understand in your previous comment. Therefore, I ended up returning mode_set_per_segment. I didn't want to repeat those lines elsewhere in the repo.

Reply:
Removed dependency from existing scripts to safely merge this PR
But in the future I strongly believe we need some sort of returns to plot modal basis

Reply:
Fully agreed 👍